src/python/pants/backend/python/subsystems/pex_build_util.py

```diff
@@ -140,40 +144,57 @@ def subsystem_dependencies(cls):
         PythonSetup,
       )
 
+    @property
+    def generate_ptex(self) -> bool:
+      return cast(bool, self.get_options().generate_ptex)
+
     @classmethod
-    def create(cls, builder, log=None):
+    def create(cls, builder, log=None, generate_ptex=None, pex_requirement=None):
       options = cls.global_instance().get_options()
       setuptools_requirement = f'setuptools=={options.setuptools_version}'
+      pex_requirement = f'pex=={pex_requirement}' if pex_requirement else 'pex'
 
       log = log or logging.getLogger(__name__)
 
       return PexBuilderWrapper(builder=builder,
                                python_repos_subsystem=PythonRepos.global_instance(),
                                python_setup_subsystem=PythonSetup.global_instance(),
                                setuptools_requirement=PythonRequirement(setuptools_requirement),
-                               log=log)
+                               pex_requirement=PythonRequirement(pex_requirement),
+                               log=log,
+                               generate_ptex=bool(generate_ptex))
 
   def __init__(self,
-               builder,
-               python_repos_subsystem,
-               python_setup_subsystem,
-               setuptools_requirement,
-               log):
-    assert isinstance(builder, PEXBuilder)
-    assert isinstance(python_repos_subsystem, PythonRepos)
-    assert isinstance(python_setup_subsystem, PythonSetup)
-    assert isinstance(setuptools_requirement, PythonRequirement)
+               builder: PEXBuilder,
+               python_repos_subsystem: PythonRepos,
+               python_setup_subsystem: PythonSetup,
+               setuptools_requirement: PythonRequirement,
+               pex_requirement: PythonRequirement,
+               log: Any,
+               generate_ptex: bool = False):
     assert log is not None
 
     self._builder = builder
     self._python_repos_subsystem = python_repos_subsystem
     self._python_setup_subsystem = python_setup_subsystem
     self._setuptools_requirement = setuptools_requirement
+    self._pex_requirement = pex_requirement
```

Comment:
As mentioned a few places - this should not be a variable, it should be the live pex version from the imports.